### PR TITLE
Add more bytesrepr benchmarks.

### PR DESCRIPTION
**Pull request by goral09**
Adds a bytesrepr bebchmarks for:
* Account
* Contract
* Contract package
* Bid
* Deploy info
* Transfer
* Era info

Closes https://github.com/casper-network/casper-node/issues/2388

### CODE REVIEW

File: types/benches/bytesrepr_bench.rs [hunk 1]
```diff
@@ -437,6 +446,300 @@ fn deserialize_u512(b: &mut Bencher) {
...
+    let contract_bytes = Contract::to_bytes(&contract).unwrap();
+    b.iter(|| Contract::from_bytes(black_box(&contract_bytes)).unwrap());
+}
+
+fn generate_contract(named_keys_len: u8, entry_points_len: u8) -> Contract {
+    let named_keys: NamedKeys = {
+        let mut tmp = BTreeMap::new();
+        (0..named_keys_len).for_each(|i| {
+            let res = tmp.insert(
+                std::str::from_utf8(&[i]).unwrap().to_string(),
```
Comment (mpapierski) [inline types/benches/bytesrepr_bench.rs]:
```suggestion
                i.to_string(),
```
or `format!("named-key-{}", i)`

File: types/benches/bytesrepr_bench.rs [hunk 2]
```diff
@@ -437,6 +446,300 @@ fn deserialize_u512(b: &mut Bencher) {
     b.iter(|| U512::from_bytes(black_box(&num_u512_bytes)))
 }
 
+fn generate_account(associated_keys_len: u8, named_keys_len: u8) -> Account {
+    let account_hash = AccountHash::default();
+    let named_keys: NamedKeys = {
+        let mut tmp = BTreeMap::new();
+        (0..named_keys_len).for_each(|i| {
```
Comment (mpapierski) [inline types/benches/bytesrepr_bench.rs]: It's used both in account generator and contract generator. You can reactor this closure into

```rust
fn sample_named_keys(n: usize) -> NamedKeys {}
```

File: types/benches/bytesrepr_bench.rs [hunk 3]
```diff
@@ -437,6 +446,300 @@ fn deserialize_u512(b: &mut Bencher) {
...
+
+fn sample_set<K: Ord, F>(fun: F, count: u8) -> BTreeSet<K>
+where
+    F: Fn(u8) -> K,
+{
+    (0..count).map(fun).collect()
+}
+
+fn sample_group(i: u8) -> Group {
+    Group::new(format!("group-{}", std::str::from_utf8(&[i]).unwrap()))
```
Comment (mpapierski) [inline types/benches/bytesrepr_bench.rs]:
```suggestion
    Group::new(format!("group-{}", i)),
```

Comment (goral09) [reply types/benches/bytesrepr_bench.rs]: Done in [`b07c708` (#2389)](https://github.com/casper-network/casper-node/pull/2389/commits/b07c70853e7d9447e20fd5a17ad635a5b8606ad5).

File: types/benches/bytesrepr_bench.rs (hunk 2 above)
Comment (goral09) [reply types/benches/bytesrepr_bench.rs]: Done in [`b07c708` (#2389)](https://github.com/casper-network/casper-node/pull/2389/commits/b07c70853e7d9447e20fd5a17ad635a5b8606ad5).

**Review by mpapierski — APPROVED** [review]

File: types/benches/bytesrepr_bench.rs (hunk 1 above)
Comment (goral09) [reply types/benches/bytesrepr_bench.rs]: Done in [`b07c708` (#2389)](https://github.com/casper-network/casper-node/pull/2389/commits/b07c70853e7d9447e20fd5a17ad635a5b8606ad5).

**Comment by goral09** [comment]
bors merge

**Pull request merged**